cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/AbstractOperations.java

```diff
@@ -165,4 +167,68 @@ public T get(String namespace, String name) {
         return operation().inNamespace(namespace).withLabels(labels).list().getItems();
     }
 
+    /**
+     * Waits until resource is in the Ready state
+     *
+     * @param namespace The namespace
+     * @param name The resource name
+     * @param timeout Timeout in {@code timeUnit}
+     * @param timeUnit Time unit
+     */
+    public Future<Void> waitUntilReady(String namespace, String name, long timeout, TimeUnit timeUnit) {
+        Future<Void> fut = Future.future();
+        log.info("Waiting for {} resource {} in namespace {} to get ready", resourceKind, name, namespace);
+        AtomicInteger calls = new AtomicInteger(0);
+        long timer = 1000L;
+        long timeoutInMs = timeUnit.toMillis(timeout);
+
+        vertx.setPeriodic(timer, timerId -> {
+            calls.addAndGet(1);
+
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                    future -> {
+                        try {
+                            R resourceOp = operation().inNamespace(namespace).withName(name);
+                            T resource = resourceOp.get();
+                            if (resource != null)   {
+                                if (Readiness.isReadinessApplicable(resource)) {
```

Comment:
I would be tempted to extract an `isReady()` method into `AbstractOperations`. In any case, I think you only need the `resourceOp` inside this block guarded by `if (Readiness.isReadinessApplicable(resource))`.